### PR TITLE
feat:  default email alert group

### PR DESCRIPTION
**Pull request by grutt**
# Description

When new users join, there is no alerting configured by default. While this is the intended behavior for things like workflow runs which may be noisy, it may be problematic for alerts for token expiration.

This change adds additional configuration settings for alerts and exposes the tenant member list as a special email group.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## What's Changed

- [x] add tenant alert setting configuration to enable workflow run failure and token expiration alerts
- [x] add tenant configuration to enable tenant member email alert group
- [x] add query to get emails for a tenant
- [x] update alerters with config
- [x] expose config on ui

### CODE REVIEW

**Comment by grutt** [comment]
image

File: internal/repository/tenant_alerting.go [hunk 1]
```diff
@@ -40,12 +44,17 @@ type TenantAlertingAPIRepository interface {
 	DeleteTenantAlertGroup(tenantId string, id string) error
 }
 
+type TenantAlertEmailGroupForSend struct {
+	TenantId pgtype.UUID `json:"tenantId"`
+	Emails   string      `json:"emails"`
```
Comment (abelanger5) [inline internal/repository/tenant_alerting.go]:
```suggestion
	Emails []string `validate:"required,dive,email,max=255"`
```

(assuming this is meant to be a list of emails) - better to validate with the string array and concat in the repository.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -134,6 +134,7 @@ model Tenant {
   rateLimits                RateLimit[]
   stepRateLimits            StepRateLimit[]
   alertEmailGroups          TenantAlertEmailGroup[]
+  alertMemberEmails         Boolean @default(true)
```
Comment (abelanger5) [inline prisma/schema.prisma:138]: I don't know at first glance what this does - could you add a comment?

File: internal/repository/prisma/tenant_alerting.go [hunk 3]
```diff
@@ -142,22 +147,48 @@ func (r *tenantAlertingEngineRepository) GetTenantAlertingSettings(ctx context.C
...
+	}
+
 	tenant, err := r.queries.GetTenantByID(ctx, tx, pgTenantId)
 
 	if err != nil {
 		return nil, err
 	}
 
+	if tenant.AlertMemberEmails {
+		emails, err := r.queries.GetMemberEmailGroup(ctx, tx, pgTenantId)
```
Comment (abelanger5) [inline internal/repository/prisma/tenant_alerting.go:174]: I actually looks like this tx is never committed, is that right? Should fix ASAP.

**Review by abelanger5 — CHANGES_REQUESTED** [review]

File: internal/repository/prisma/tenant_alerting.go (hunk 3 above)
Comment (grutt) [reply internal/repository/prisma/tenant_alerting.go:174]: good catch

**Review by abelanger5 — APPROVED** [review]

**Pull request merged**